### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by acauret**
Update docs after tflint fix

### CODE REVIEW

**Comment by segraef** [comment]
```[tasklist]
### Pull Request Review Checklist
- [ ] [ID: TFFR1 - Category: Composition - Cross-Referencing Modules](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr1---category-composition---cross-referencing-modules)
- [ ] [ID: TFFR2 - Category: Outputs - Additional Terraform Outputs](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr2---category-outputs---additional-terraform-outputs)
- [ ] [ID: TFNFR1 - Category: Documentation - Descriptions](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr1---category-documentation---descriptions)
- [ ] [ID: TFNFR2 - Category: Documentation - Module Documentation Generation](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr2---category-documentation---module-documentation-generation)
- [ ] [ID: TFNFR3 - Category: Contribution/Support - GitHub Repo Branch Protection](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr3---category-contributionsupport---github-repo-branch-protection)
- [ ] [ID: TFNFR4 - Category: Composition - Code Styling - lower snake_casing](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr4---category-composition---code-styling---lower-snake_casing)
- [ ] [ID: TFNFR5 - Category: Testing - Test Tooling](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr5---category-testing---test-tooling)
- [ ] [ID: TFNFR6 - Category: Code Style - The Order of resource and data in the Same File](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr6---category-code-style---the-order-of-resource-and-data-in-the-same-file)
- [ ] [ID: TFNFR7 - Category: Code Style - The Use of count and for_each](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr7---category-code-style---the-use-of-count-and-for_each)
- [ ] [ID: TFNFR8 - Category: Code Style - Orders Within resource and data Blocks](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr8---category-code-style---orders-within-resource-and-data-blocks)
- [ ] [ID: TFNFR9 - Category: Code Style - Order within a module block](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr9---category-code-style---order-within-a-module-block)
- [ ] [ID: TFNFR10 - Category: Code Style - Values in ignore_changes passed to provider, depends_on, lifecycle blocks are not allowed to use double quotations](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr10---category-code-style---values-in-ignore_changes-passed-to-provider-depends_on-lifecycle-blocks-are-not-allowed-to-use-double-quotations)
- [ ] [ID: TFNFR11 - Category: Code Style - null comparison as creation toogle](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr11---category-code-style---null-comparison-as-creation-toogle)
- [ ] [ID: TFNFR12 - Category: Code Style - Optional nested object argument should use dynamic](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr12---category-code-style---optional-nested-object-argument-should-use-dynamic)
- [ ] [ID: TFNFR13 - Category: Code Style - Use coalesce or try when setting default values for nullable expressions](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr13---category-code-style---use-coalesce-or-try-when-setting-default-values-for-nullable-expressions)
- [ ] [ID: TFFR14 - Category: Inputs - No enabled or module_depends_on variable](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr14---category-inputs---no-enabled-or-module_depends_on-variable)
- [ ] [ID: TFNFR15 - Category: Code Style - Order to define variable](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr15---category-code-style---order-to-define-variable)
- [ ] [ID: TFNFR16 - Category: Code Style - Name of a variable MUST follow rules](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr16---category-code-style---name-of-a-variable-must-follow-rules)
- [ ] [ID: TFNFR17 - Category: Code Style - Every variable MUST come with a description](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr17---category-code-style---every-variable-must-come-with-a-description)
- [ ] [ID: TFNFR18 - Category: Code Style - Every variable MUST have an appropriate type](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr18---category-code-style---every-variable-must-have-an-appropriate-type)
- [ ] [ID: TFNFR19 - Category: Code Style - variable containing confidential data should be declared as sensitive = true](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr19---category-code-style---variable-containing-confidential-data-should-be-declared-as-sensitive--true)
- [ ] [ID: TFNFR20 - Category: Code Style - Declare nullable = false when it’s possible](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr20---category-code-style---declare-nullable--false-when-its-possible)
- [ ] [ID: TFNFR21 - Category: Code Style - MUST NOT declare nullable = true](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr21---category-code-style---must-not-declare-nullable--true)
- [ ] [ID: TFNFR22 - Category: Code Style - MUST NOT declare sensitive = false](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr22---category-code-style---must-not-declare-sensitive--false)
- [ ] [ID: TFNFR23 - Category: Code Style - variable with sensitive = true MUST NOT have default value](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr23---category-code-style---variable-with-sensitive--true-must-not-have-default-value-unless-the-default-value-represents-turning-off-a-feature-like-default--null-or-default--)
- [ ] [ID: TFNFR24 - Category: Code Style - Deal with deprecated variable](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr24---category-code-style---deal-with-deprecated-variable)
- [ ] [ID: TFNFR25 - Category: Code Style - All verified modules MUST have terraform.tf file and required_version MUST be set](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr25---category-code-style---all-verified-modules-must-have-terraformtf-file-and-required_version-must-be-set)
- [ ] [ID: TFNFR26 - Category: Code Style - Providers MUST be declared in the required_providers block in terraform.tf and MUST have a constraint on minimum and maximum major version](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr26---category-code-style---providers-must-be-declared-in-the-required_providers-block-in-terraformtf-and-must-have-a-constraint-on-minimum-and-maximum-major-version)
- [ ] [ID: TFNFR27 - Category: Code Style - Declaration of a provider in the module](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr27---category-code-style---declaration-of-a-provider-in-the-module)
- [ ] [ID: TFNFR28 - Category: Code Style - output MUST be arranged alphabetically](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr28---category-code-style---output-must-be-arranged-alphabetically)
- [ ] [ID: TFNFR29 - Category: Code Style - output contains confidential data should declare sensitive = true](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr29---category-code-style---output-contains-confidential-data-should-declare-sensitive--true)
- [ ] [ID: TFNFR30 - Category: Code Style - Dealing with Deprecated outputs](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr30---category-code-style---dealing-with-deprecated-outputs)
- [ ] [ID: TFNFR31 - Category: Code Style - locals.tf MUST contain only locals blocks](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr31---category-code-style---localstf-must-contain-only-locals-blocks)
- [ ] [ID: TFNFR32 - Category: Code Style - local should be arranged alphabetically](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr32---category-code-style---local-should-be-arranged-alphabetically)
- [ ] [ID: TFNFR33 - Category: Code Style - local should use types as precise as possible](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr33---category-code-style---local-should-use-types-as-precise-as-possible)
- [ ] [ID: TFNFR34 - Category: Code Style - Feature toggle MUST be used to ensure forward compatibility of versions and avoid unexpected changes caused by upgrades](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr34---category-code-style---feature-toggle-must-be-used-to-ensure-forward-compatibility-of-versions-and-avoid-unexpected-changes-caused-by-upgrades)
- [ ] [ID: TFNFR35 - Category: Code Style - Changes that might be breaking change MUST be reviewed with caution](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr35---category-code-style---changes-that-might-be-breaking-change-must-be-reviewed-with-caution)
- [ ] [ID: TFNFR36 - Category: Code Style - Example code MUST set prevent_deletion_if_contains_resources to false in provider block](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr36---category-code-style---example-code-must-set-prevent_deletion_if_contains_resources-to-false-in-provider-block)
- [ ] [ID: TFNFR37 - Category: Code Style - Module owner MAY use tools like newres](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr37---category-code-style---module-owner-may-use-tools-like-newreshttpsgithubcomlonegunmanbnewres)
```

**Comment by segraef** [comment]
| Specification | Review |
|--- | --- |
| [ID: TFFR1 - Category: Composition - Cross-Referencing Modules](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr1---category-composition---cross-referencing-modules) | <li>[ ] OK</li> |
| [ID: TFFR2 - Category: Outputs - Additional Terraform Outputs](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr2---category-outputs---additional-terraform-outputs) | <li>[ ] OK</li> |
| [ID: TFNFR1 - Category: Documentation - Descriptions](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr1---category-documentation---descriptions) | <li>[ ] OK</li> |
| [ID: TFNFR2 - Category: Documentation - Module Documentation Generation](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr2---category-documentation---module-documentation-generation) | <li>[ ] OK</li> |
| [ID: TFNFR3 - Category: Contribution/Support - GitHub Repo Branch Protection](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr3---category-contributionsupport---github-repo-branch-protection) | <li>[ ] OK</li> |
| [ID: TFNFR4 - Category: Composition - Code Styling - lower snake_casing](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr4---category-composition---code-styling---lower-snake_casing) | <li>[ ] OK</li> |
| [ID: TFNFR5 - Category: Testing - Test Tooling](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr5---category-testing---test-tooling) | <li>[ ] OK</li> |
| [ID: TFNFR6 - Category: Code Style - The Order of resource and data in the Same File](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr6---category-code-style---the-order-of-resource-and-data-in-the-same-file) | <li>[ ] OK</li> |
| [ID: TFNFR7 - Category: Code Style - The Use of count and for_each](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr7---category-code-style---the-use-of-count-and-for_each) | <li>[ ] OK</li> |
| [ID: TFNFR8 - Category: Code Style - Orders Within resource and data Blocks](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr8---category-code-style---orders-within-resource-and-data-blocks) | <li>[ ] OK</li> |
| [ID: TFNFR9 - Category: Code Style - Order within a module block](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr9---category-code-style---order-within-a-module-block) | <li>[ ] OK</li> |
| [ID: TFNFR10 - Category: Code Style - Values in ignore_changes passed to provider, depends_on, lifecycle blocks are not allowed to use double quotations](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr10---category-code-style---values-in-ignore_changes-passed-to-provider-depends_on-lifecycle-blocks-are-not-allowed-to-use-double-quotations) | <li>[ ] OK</li> |
| [ID: TFNFR11 - Category: Code Style - null comparison as creation toogle](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr11---category-code-style---null-comparison-as-creation-toogle) | <li>[ ] OK</li> |
| [ID: TFNFR12 - Category: Code Style - Optional nested object argument should use dynamic](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr12---category-code-style---optional-nested-object-argument-should-use-dynamic) | <li>[ ] OK</li> |
| [ID: TFNFR13 - Category: Code Style - Use coalesce or try when setting default values for nullable expressions](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr13---category-code-style---use-coalesce-or-try-when-setting-default-values-for-nullable-expressions) | <li>[ ] OK</li> |
| [ID: TFFR14 - Category: Inputs - No enabled or module_depends_on variable](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr14---category-inputs---no-enabled-or-module_depends_on-variable) | <li>[ ] OK</li> |
| [ID: TFNFR15 - Category: Code Style - Order to define variable](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr15---category-code-style---order-to-define-variable) | <li>[ ] OK</li> |
| [ID: TFNFR16 - Category: Code Style - Name of a variable MUST follow rules](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr16---category-code-style---name-of-a-variable-must-follow-rules) | <li>[ ] OK</li> |
| [ID: TFNFR17 - Category: Code Style - Every variable MUST come with a description](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr17---category-code-style---every-variable-must-come-with-a-description) | <li>[ ] OK</li> |
| [ID: TFNFR18 - Category: Code Style - Every variable MUST have an appropriate type](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr18---category-code-style---every-variable-must-have-an-appropriate-type) | <li>[ ] OK</li> |
| [ID: TFNFR19 - Category: Code Style - variable containing confidential data should be declared as sensitive = true](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr19---category-code-style---variable-containing-confidential-data-should-be-declared-as-sensitive--true) | <li>[ ] OK</li> |
| [ID: TFNFR20 - Category: Code Style - Declare nullable = false when it’s possible](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr20---category-code-style---declare-nullable--false-when-its-possible) | <li>[ ] OK</li> |
| [ID: TFNFR21 - Category: Code Style - MUST NOT declare nullable = true](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr21---category-code-style---must-not-declare-nullable--true) | <li>[ ] OK</li> |
| [ID: TFNFR22 - Category: Code Style - MUST NOT declare sensitive = false](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr22---category-code-style---must-not-declare-sensitive--false) | <li>[ ] OK</li> |
| [ID: TFNFR23 - Category: Code Style - variable with sensitive = true MUST NOT have default value](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr23---category-code-style---variable-with-sensitive--true-must-not-have-default-value-unless-the-default-value-represents-turning-off-a-feature-like-default--null-or-default--) | <li>[ ] OK</li> |
| [ID: TFNFR24 - Category: Code Style - Deal with deprecated variable](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr24---category-code-style---deal-with-deprecated-variable) | <li>[ ] OK</li> |
| [ID: TFNFR25 - Category: Code Style - All verified modules MUST have terraform.tf file and required_version MUST be set](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr25---category-code-style---all-verified-modules-must-have-terraformtf-file-and-required_version-must-be-set) | <li>[ ] OK</li> |
| [ID: TFNFR26 - Category: Code Style - Providers MUST be declared in the required_providers block in terraform.tf and MUST have a constraint on minimum and maximum major version](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr26---category-code-style---providers-must-be-declared-in-the-required_providers-block-in-terraformtf-and-must-have-a-constraint-on-minimum-and-maximum-major-version) | <li>[ ] OK</li> |
| [ID: TFNFR27 - Category: Code Style - Declaration of a provider in the module](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr27---category-code-style---declaration-of-a-provider-in-the-module) | <li>[ ] OK</li> |
| [ID: TFNFR28 - Category: Code Style - output MUST be arranged alphabetically](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr28---category-code-style---output-must-be-arranged-alphabetically) | <li>[ ] OK</li> |
| [ID: TFNFR29 - Category: Code Style - output contains confidential data should declare sensitive = true](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr29---category-code-style---output-contains-confidential-data-should-declare-sensitive--true) | <li>[ ] OK</li> |
| [ID: TFNFR30 - Category: Code Style - Dealing with Deprecated outputs](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr30---category-code-style---dealing-with-deprecated-outputs) | <li>[ ] OK</li> |
| [ID: TFNFR31 - Category: Code Style - locals.tf MUST contain only locals blocks](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr31---category-code-style---localstf-must-contain-only-locals-blocks) | <li>[ ] OK</li> |
| [ID: TFNFR32 - Category: Code Style - local should be arranged alphabetically](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr32---category-code-style---local-should-be-arranged-alphabetically) | <li>[ ] OK</li> |
| [ID: TFNFR33 - Category: Code Style - local should use types as precise as possible](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr33---category-code-style---local-should-use-types-as-precise-as-possible) | <li>[ ] OK</li> |
| [ID: TFNFR34 - Category: Code Style - Feature toggle MUST be used to ensure forward compatibility of versions and avoid unexpected changes caused by upgrades](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr34---category-code-style---feature-toggle-must-be-used-to-ensure-forward-compatibility-of-versions-and-avoid-unexpected-changes-caused-by-upgrades) | <li>[ ] OK</li> |
| [ID: TFNFR35 - Category: Code Style - Changes that might be breaking change MUST be reviewed with caution](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr35---category-code-style---changes-that-might-be-breaking-change-must-be-reviewed-with-caution) | <li>[ ] OK</li> |
| [ID: TFNFR36 - Category: Code Style - Example code MUST set prevent_deletion_if_contains_resources to false in provider block](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr36---category-code-style---example-code-must-set-prevent_deletion_if_contains_resources-to-false-in-provider-block) | <li>[ ] OK</li> |
| [ID: TFNFR37 - Category: Code Style - Module owner MAY use tools like newres](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr37---category-code-style---module-owner-may-use-tools-like-newreshttpsgithubcomlonegunmanbnewres) | <li>[ ] OK</li> |

**Comment by segraef** [comment]
| Specification | Review |
|--- | --- |
| [ID: TFFR1 - Category: Composition - Cross-Referencing Modules](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr1---category-composition---cross-referencing-modules) | <li>[ ] OK</li> |
| [ID: TFFR2 - Category: Outputs - Additional Terraform Outputs](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr2---category-outputs---additional-terraform-outputs) | <li>[ ] OK</li> |
| [ID: TFNFR1 - Category: Documentation - Descriptions](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr1---category-documentation---descriptions) | <li>[ ] OK</li> |
| [ID: TFNFR2 - Category: Documentation - Module Documentation Generation](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr2---category-documentation---module-documentation-generation) | <li>[ ] OK</li> |
| [ID: TFNFR3 - Category: Contribution/Support - GitHub Repo Branch Protection](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr3---category-contributionsupport---github-repo-branch-protection) | <li>[ ] OK</li> |
| [ID: TFNFR4 - Category: Composition - Code Styling - lower snake_casing](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr4---category-composition---code-styling---lower-snake_casing) | <li>[ ] OK</li> |
| [ID: TFNFR5 - Category: Testing - Test Tooling](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr5---category-testing---test-tooling) | <li>[ ] OK</li> |
| [ID: TFNFR6 - Category: Code Style - The Order of resource and data in the Same File](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr6---category-code-style---the-order-of-resource-and-data-in-the-same-file) | <li>[ ] OK</li> |
| [ID: TFNFR7 - Category: Code Style - The Use of count and for_each](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr7---category-code-style---the-use-of-count-and-for_each) | <li>[ ] OK</li> |
| [ID: TFNFR8 - Category: Code Style - Orders Within resource and data Blocks](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr8---category-code-style---orders-within-resource-and-data-blocks) | <li>[ ] OK</li> |
| [ID: TFNFR9 - Category: Code Style - Order within a module block](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr9---category-code-style---order-within-a-module-block) | <li>[ ] OK</li> |
| [ID: TFNFR10 - Category: Code Style - Values in ignore_changes passed to provider, depends_on, lifecycle blocks are not allowed to use double quotations](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr10---category-code-style---values-in-ignore_changes-passed-to-provider-depends_on-lifecycle-blocks-are-not-allowed-to-use-double-quotations) | <li>[ ] OK</li> |
| [ID: TFNFR11 - Category: Code Style - null comparison as creation toogle](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr11---category-code-style---null-comparison-as-creation-toogle) | <li>[ ] OK</li> |
| [ID: TFNFR12 - Category: Code Style - Optional nested object argument should use dynamic](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr12---category-code-style---optional-nested-object-argument-should-use-dynamic) | <li>[ ] OK</li> |
| [ID: TFNFR13 - Category: Code Style - Use coalesce or try when setting default values for nullable expressions](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr13---category-code-style---use-coalesce-or-try-when-setting-default-values-for-nullable-expressions) | <li>[ ] OK</li> |
| [ID: TFFR14 - Category: Inputs - No enabled or module_depends_on variable](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr14---category-inputs---no-enabled-or-module_depends_on-variable) | <li>[ ] OK</li> |
| [ID: TFNFR15 - Category: Code Style - Order to define variable](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr15---category-code-style---order-to-define-variable) | <li>[ ] OK</li> |
| [ID: TFNFR16 - Category: Code Style - Name of a variable MUST follow rules](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr16---category-code-style---name-of-a-variable-must-follow-rules) | <li>[ ] OK</li> |
| [ID: TFNFR17 - Category: Code Style - Every variable MUST come with a description](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr17---category-code-style---every-variable-must-come-with-a-description) | <li>[ ] OK</li> |
| [ID: TFNFR18 - Category: Code Style - Every variable MUST have an appropriate type](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr18---category-code-style---every-variable-must-have-an-appropriate-type) | <li>[ ] OK</li> |
| [ID: TFNFR19 - Category: Code Style - variable containing confidential data should be declared as sensitive = true](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr19---category-code-style---variable-containing-confidential-data-should-be-declared-as-sensitive--true) | <li>[ ] OK</li> |
| [ID: TFNFR20 - Category: Code Style - Declare nullable = false when it’s possible](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr20---category-code-style---declare-nullable--false-when-its-possible) | <li>[ ] OK</li> |
| [ID: TFNFR21 - Category: Code Style - MUST NOT declare nullable = true](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr21---category-code-style---must-not-declare-nullable--true) | <li>[ ] OK</li> |
| [ID: TFNFR22 - Category: Code Style - MUST NOT declare sensitive = false](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr22---category-code-style---must-not-declare-sensitive--false) | <li>[ ] OK</li> |
| [ID: TFNFR23 - Category: Code Style - variable with sensitive = true MUST NOT have default value](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr23---category-code-style---variable-with-sensitive--true-must-not-have-default-value-unless-the-default-value-represents-turning-off-a-feature-like-default--null-or-default--) | <li>[ ] OK</li> |
| [ID: TFNFR24 - Category: Code Style - Deal with deprecated variable](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr24---category-code-style---deal-with-deprecated-variable) | <li>[ ] OK</li> |
| [ID: TFNFR25 - Category: Code Style - All verified modules MUST have terraform.tf file and required_version MUST be set](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr25---category-code-style---all-verified-modules-must-have-terraformtf-file-and-required_version-must-be-set) | <li>[ ] OK</li> |
| [ID: TFNFR26 - Category: Code Style - Providers MUST be declared in the required_providers block in terraform.tf and MUST have a constraint on minimum and maximum major version](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr26---category-code-style---providers-must-be-declared-in-the-required_providers-block-in-terraformtf-and-must-have-a-constraint-on-minimum-and-maximum-major-version) | <li>[ ] OK</li> |
| [ID: TFNFR27 - Category: Code Style - Declaration of a provider in the module](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr27---category-code-style---declaration-of-a-provider-in-the-module) | <li>[ ] OK</li> |
| [ID: TFNFR28 - Category: Code Style - output MUST be arranged alphabetically](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr28---category-code-style---output-must-be-arranged-alphabetically) | <li>[ ] OK</li> |
| [ID: TFNFR29 - Category: Code Style - output contains confidential data should declare sensitive = true](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr29---category-code-style---output-contains-confidential-data-should-declare-sensitive--true) | <li>[ ] OK</li> |
| [ID: TFNFR30 - Category: Code Style - Dealing with Deprecated outputs](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr30---category-code-style---dealing-with-deprecated-outputs) | <li>[ ] OK</li> |
| [ID: TFNFR31 - Category: Code Style - locals.tf MUST contain only locals blocks](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr31---category-code-style---localstf-must-contain-only-locals-blocks) | <li>[ ] OK</li> |
| [ID: TFNFR32 - Category: Code Style - local should be arranged alphabetically](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr32---category-code-style---local-should-be-arranged-alphabetically) | <li>[ ] OK</li> |
| [ID: TFNFR33 - Category: Code Style - local should use types as precise as possible](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr33---category-code-style---local-should-use-types-as-precise-as-possible) | <li>[ ] OK</li> |
| [ID: TFNFR34 - Category: Code Style - Feature toggle MUST be used to ensure forward compatibility of versions and avoid unexpected changes caused by upgrades](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr34---category-code-style---feature-toggle-must-be-used-to-ensure-forward-compatibility-of-versions-and-avoid-unexpected-changes-caused-by-upgrades) | <li>[ ] OK</li> |
| [ID: TFNFR35 - Category: Code Style - Changes that might be breaking change MUST be reviewed with caution](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr35---category-code-style---changes-that-might-be-breaking-change-must-be-reviewed-with-caution) | <li>[ ] OK</li> |
| [ID: TFNFR36 - Category: Code Style - Example code MUST set prevent_deletion_if_contains_resources to false in provider block](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr36---category-code-style---example-code-must-set-prevent_deletion_if_contains_resources-to-false-in-provider-block) | <li>[ ] OK</li> |
| [ID: TFNFR37 - Category: Code Style - Module owner MAY use tools like newres](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr37---category-code-style---module-owner-may-use-tools-like-newreshttpsgithubcomlonegunmanbnewres) | <li>[ ] OK</li> |

**Comment by segraef** [comment]
going to review the whole repository with @mbilalamjad tomorrow

File: examples/customer-managed-key/main.tf [hunk 1]
```diff
@@ -1,6 +1,10 @@
 terraform {
   required_version = ">= 1.0.0"
   required_providers {
+    azuread = {
+      source  = "hashicorp/azuread"
+      version = ">= 2.15.0"
```
Comment (segraef) [inline examples/customer-managed-key/main.tf:6]: version constraint missing see https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr26---category-code-style---providers-must-be-declared-in-the-required_providers-block-in-terraformtf-and-must-have-a-constraint-on-minimum-and-maximum-major-version

Comment (segraef) [reply examples/customer-managed-key/main.tf:6]: As per [TFNFR36](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr36---category-code-style---example-code-must-set-prevent_deletion_if_contains_resources-to-false-in-provider-block) set `prevent_deletion_if_contains_resources` in `providers`:
```terraform
resource_group {
      prevent_deletion_if_contains_resources = true
    }
```

File: examples/private-endpoint/main.tf [hunk 2]
```diff
@@ -17,6 +17,7 @@ provider "azurerm" {
 }
 
 # We need the tenant id.
+# tflint-ignore: terraform_unused_declarations
```
Comment (segraef) [inline examples/private-endpoint/main.tf:20]: As per [TFNFR36](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr36---category-code-style---example-code-must-set-prevent_deletion_if_contains_resources-to-false-in-provider-block) set `prevent_deletion_if_contains_resources` in `providers`:
```terraform
resource_group {
      prevent_deletion_if_contains_resources = true
    }
```

**Review by segraef — CHANGES_REQUESTED** [review]
some of the requested changes are not raised in this PR but should be changed in your files in main, thanks

File: examples/private-endpoint/main.tf (hunk 2 above)
Comment (acauret) [reply examples/private-endpoint/main.tf:20]: Updated as requested although this should be false

**Review by mbilalamjad — DISMISSED** [review]

**Comment by segraef** [comment]
@acauret @mbilalamjad was so kind and did the AVM-Review https://github.com/Azure/terraform-azurerm-avm-res-databricks-workspace/issues/20, please sort the open items so we can publish your module, thanks.

**Comment by segraef** [comment]
> @acauret @mbilalamjad was so kind and did the AVM-Review #20, please sort the open items so we can publish your module, thanks.

Hi @acauret , did you find the time to look into it?

**Review by segraef — APPROVED** [review]
🤙🏽

**Pull request merged**